tests/autoyast/installation.pm

```diff
@@ -235,6 +236,11 @@ sub run {
             wait_screen_change { send_key $cmd{ok} };
             @needles = grep { $_ ne 'inst-betawarning' } @needles;
             push(@needles, 'autoyast-license') if $expected_licenses;
+            if (my $disk = get_var('AUTOYAST_PRE_ENCRYPT_DISK')) {
```

Comment:
As this module becomes monster, have you considered to have separate module which does disk pre-partitioning?
Also, keep in mind that beta warning is not there for GM builds, so we should not use it for sync here. So I would say we should use `confirm` option (see https://documentation.suse.com/sles/15-SP1/single-html/SLES-autoyast/#CreateProfile-General-mode) to guarantee that we can modify system before actual installation starts.

Reply:
It is a monster loop with not fixed expectations, I agree but it would require a lot of work to break it into pieces. I have in mind for future. That sync point is just what I was looking for, thanks. I will fix it.

Reply:
Installation Settings screen is too late to format the disk: http://rivera-workstation.suse.cz/tests/1010#step/installation/5, I cannot see any other place, because even if we create a new needle specific for this, it might match first the one for the beta warning.

Reply:
I created a separated module and chose startshell as synchronization point.